link.go

```diff
@@ -284,35 +288,64 @@ func (l *link) setSettleModes(resp *performAttach) error {
 	return nil
 }
 
-func (l *link) mux() {
-	defer l.muxDetach()
-
+func (l *link) doFlow() (ok bool, enableOutgoingTransfers bool) {
```

Comment:
It would be good to add a comment explaining the meaning of `ok` and `enableOutgoingTransfers`

Reply:
Agreed, will add.

It's gross because it's a lift and shift of the older code, which intermixed sender and receiver checks. Didn't want to change too much.

Reply:
Conversely, would it make sense to change this to return some descriptive enum instead (assuming an enum can cover the permutations)?

Reply:
Thinking about this more, can we wait on changing it? The simplest thing that would "fix it" would be to break apart the sender vs receiver flows and I know we have that planned separately anyways.

Reply:
Comment added. It's gross, but we'll get rid of it soon.